sys/vm/vm_cheri_revoke.c

```diff
@@ -677,12 +677,13 @@ vm_cheri_revoke_object_at(const struct vm_cheri_revoke_cookie *crc,
 				CHERI_REVOKE_STATS_INC(crst, pages_skip_fast,
```

Comment:
So I guess vm_cheri_revoke_skip_fault(obj) doesn't care about this page since it doesn't accept the page index (`ipi`), it's just a "is this object using the swap pager so we can try to optimize", and the case you are fixing here is "we can't optimize because the page we are currently trying to scan needs to be faulted in".  I had misread the code incorrectly the first time (I was assuming nextpindex must always be "after" `ipi` which seems to be the same assumption that led to the bug you are fixing now).

Reply:
That's exactly right.